scripts/deploy.sh

```diff
@@ -6,6 +6,14 @@ SERVICE_NAME=""
 AWS_CLUSTER_NAME=""
 AWS_RESOURCE_NAME=""
 TAG=""
+
+# validate if $1 variable exists if not, it means it is a tagged build, without circle branch variable
```

Comment:
This doesn't seem so robust

Reply:
@fredericojordan I'm seeing some possibilities to rewrite this but do you have a suggestion for this one?

Reply:
Instead of simply calling the script with the `$CIRCLE_BRANCH` env var:

```
bash scripts/deploy.sh $CIRCLE_BRANCH
```

there are multiple [CircleCI env vars](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) we could take advantage of, including `CIRCLE_TAG`:

> The name of the git tag, if the current build is tagged. For more information, see the [Git Tag Job Execution](https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag).

Reply:
@rkpinheiro Instead of simply calling the script with the `$CIRCLE_BRANCH` env var:

https://github.com/CheesecakeLabs/django-drf-boilerplate/blob/8ac0a346fd265855fd2ab8c3339b8ba6ed5fc9df/.circleci/config.yml#L102

there are multiple [CircleCI env vars](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) we could take advantage of, including `CIRCLE_TAG`:

> The name of the git tag, if the current build is tagged. For more information, see the [Git Tag Job Execution](https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag).

Reply:
@fredericojordan we need to use the `$CIRCLE_BRANCH` env var because our script uses it for deploying on lab and on staging. Our problem is that we need the `$CIRCLE_TAG` when deploying on prod.

What do you think about calling the script with both env vars and then checking against them on the script itself?